base/compiler/tfuncs.jl

```diff
@@ -301,6 +301,7 @@ function sizeof_nothrow(@nospecialize(x))
         return true
     end
     isconstType(x) && (x = x.parameters[1])
+    x = widenconst(x)
```

Comment:
should this be in the else block (although `widenconst` is idempotent so it shouldn't matter)